### PR TITLE
refactor: ReleaseResult

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Much refactoring and consolidation of the Release Result type.

## Context:

Maintainability

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/helm/index.ts [hunk 1]
```diff
@@ -23,12 +23,16 @@ export const defaultConfig = {
   },
 };
 
+export interface RepositoryData extends ReleaseResult {
+  name: string;
```
Comment (viceice) [inline lib/datasource/helm/index.ts:27]: For what we need this?

File: lib/datasource/npm/get.ts [hunk 2]
```diff
@@ -35,7 +35,6 @@ export interface NpmDependency extends ReleaseResult {
   deprecationSource?: string;
   name: string;
   homepage: string;
-  latestVersion: string;
```
Comment (viceice) [inline lib/datasource/npm/get.ts:38]: I think this is used by internal tools

File: lib/datasource/npm/releases.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ export async function getReleases({
   }
   const res = await getDependency(lookupName);
   if (res) {
-    res.tags = res['dist-tags'];
+    res.releaseTags = res['dist-tags'];
```
Comment (viceice) [inline lib/datasource/npm/releases.ts:14]: What the cause for this?

**Review by viceice — COMMENTED** [review]
Completing tomorrow

File: lib/datasource/helm/index.ts (hunk 1 above)
Comment (rarkins) [reply lib/datasource/helm/index.ts:27]: The cache is for the whole helm repo, so it does a `find` based on `name`. It should be refactored from being an array of packages to being a Record

File: lib/datasource/npm/get.ts (hunk 2 above)
Comment (rarkins) [reply lib/datasource/npm/get.ts:38]: It's only fetched for `npm`. Can you refactor to be `res.tags.latest` ?

File: lib/datasource/npm/releases.ts (hunk 3 above)
Comment (rarkins) [reply lib/datasource/npm/releases.ts:14]: Purely opinionated - I thought `tags` was too generic. I can revert though

File: lib/datasource/helm/index.ts (hunk 1 above)
Comment (rarkins) [reply lib/datasource/helm/index.ts]: Done

**Pull request closed without merging**